includes/class-wc-shortcodes.php

```diff
@@ -49,6 +49,71 @@ public static function init() {
 
 		// Alias for pre 2.1 compatibility.
 		add_shortcode( 'woocommerce_messages', __CLASS__ . '::shop_messages' );
+
+		if ( function_exists( 'register_block_type' ) ) {
+			foreach ( $shortcodes as $shortcode => $function ) {
+				register_block_type( 'woocommerce/' . str_replace( '_', '-', $shortcode ), array(
+					'render_callback' => $function
+				) );
+			}
+		}
+		add_action( 'enqueue_block_editor_assets', array( __CLASS__, 'enqueue_block_editor_assets' ) );
+	}
+
+	/**
+	 * Handle the assets registration for the Gutenberg Editor
+	 */
+	public static function enqueue_block_editor_assets() {
+		wp_register_script(
+			'woocommerce-editor-blocks',
+			plugins_url( 'assets/js/admin/editor-blocks.js', WC_PLUGIN_FILE ),
+			array( 'wp-blocks', 'wp-element' )
+		);
+		wp_enqueue_script( 'woocommerce-editor-blocks' );
+
+		$show_all = false;
+		$qty_products = wp_count_posts( 'product' )->publish;
+		if ( $qty_products < 10 ) {
+			$products = wc_get_products( array( 'post_status' => 'publish', 'limit' => 10 ) );
+			foreach ( $products as $product ) {
+				$product_arr = array(
+					'id' => $product->get_id(),
+					'name' => $product->get_name(),
+					'price' => $product->get_price(),
+					'images' => array(),
+				);
+				$image_arr = wp_get_attachment_image_src( $product->get_image_id() );
+				if ( $image_arr ) {
+					$product_arr['images'][] = array_combine( array(
+						'src',
+						'width',
+						'height',
+						'is_intermediate',
+					), $image_arr );
+				}
+				$show_all[] = $product_arr;
+			}
+		}
+
+		wp_localize_script( 'woocommerce-editor-blocks', 'wcEditorBlocksI18n', array(
+			'rest' => array(
+				'url' => rest_url(),
+				'nonce' => wp_create_nonce( 'wp_rest' ),
+			),
+			'currency_symbol' => get_woocommerce_currency_symbol(),
+			'show_all' => $show_all,
+			'strings' => array(
+				'Product' => __( 'Product', 'woocommerce' ),
+				'Search Products' => __( 'Search Products', 'woocommerce' ),
+				'Search products…' => __( 'Search products…', 'woocommerce' ),
+				'No products found.' => __( 'No products found.', 'woocommerce' ),
+				'Select a Product' => __( 'Select a Product', 'woocommerce' ),
+				'Buy Now' => __( 'Buy Now', 'woocommerce' ),
+				'Loading…' => __( 'Loading…', 'woocommerce' ),
```

Comment:
Can we change to `&hellip;`?

Reply:
Annoyingly, no -- it would output on the page as `&hellip` instead of `…`